app/Http/Controllers/ConfigController.php

```diff
@@ -38,31 +39,39 @@ public function handleConfiguration(Request $request)
         $configRow = Config::where('store_id', $storeId)->first();
         if (isset($configRow['id'])) {
             $config = Config::find($configRow['id']);
-            $config->active = isset($activeStatus) ? true : false;
-            $config->store_id = $storeId;
-            $config->shopkey = $shopkey;
-            $config->save();
-        } else {
-            $config = new Config;
-            $config->active = isset($activeStatus) ? true : false;
-            $config->store_id = $storeId;
-            $config->shopkey = $shopkey;
-            $config->save();
         }
+        $config = isset($config) ? $config : new Config();
+        $config->active = isset($activeStatus) ? true : false;
+        $config->store_id = $storeId;
+        $config->shopkey = $shopkey;
+        $config->save();
+
+        $this->storeToSession([
+            'access_token' => $accessToken,
+            'context' => $context,
+            'store_hash' => $storeHash,
+            'saved' => true
+        ]);
 
-        Session::put('access_token', $accessToken);
-        Session::put('context', $context);
-        Session::put('store_hash', $storeHash);
-        Session::put('saved', true);
+        $this->addSnippetToStore($storeHash, $config);
 
+        return view('app', [
+            'shopkey' => $shopkey,
+            'active_status' => $activeStatus
+        ]);
+    }
+
+    private function addSnippetToStore(string $storeHash, Config $config)
+    {
         $xmlResponse = $this->makeBigCommerceAPIRequest(
             'GET',
             'stores/' . $this->getStoreHash() . '/v2/store'
         );
+
         $collection = $this->createCollectionFromXml($xmlResponse);
 
         if (isset($collection['features']['stencil_enabled'])) {
-            // Deleting old script when ever someone save settings
+            // Delete any previously added script
```

Comment:
Instead of adding a comment, describing what a code block does, why not put this into a separate method?